File: day_1/solution.py
with open("./input.txt", "r") as input:
    input_int = list(map(int, input.readlines()))

# pt1

t_increases = sum([1 if input_int[i-1] < input_int[i] else 0 for i in range(len(input_int))])

print(f"Part 1: {t_increases}")

# pt2

sliding = [input_int[i-1] + input_int[i] + input_int[i+1] for i in range(len(input_int)-1)]
t_increases_s = sum([1 if sliding[i-1] < sliding[i] else 0 for i in range(len(sliding))])

print(f"Part 2: {t_increases_s}")

File: day_1/go/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

// file, _ := os.Open("./day_01/input.txt")
// defer file.Close()

// scanner := bufio.NewScanner(file)
// var records []int
// for scanner.Scan() {
// 	n, _ := strconv.Atoi(scanner.Text())
// 	records = append(records, n)

func parse(instr string) ([]int, error) {
	var out []int
	instr = strings.TrimSpace(instr)
	for _, val := range strings.Split(instr, "\n") {
		n, err := strconv.Atoi(val)
		if err != nil {
			return nil, err
		}
		out = append(out, n)
	}
	return out, nil
}

func solve(input []int, wSize int) int {
	count := 0

	for i := wSize; i < len(input); i++ {
		if input[i] > input[i-wSize] {
			count += 1
		}
	}

	return count
}

func main() {
	f, _ := os.ReadFile("./day_1/input.txt")
	input, _ := parse(string(f))

	fmt.Println("Part 1:", solve(input, 1))
	fmt.Println("Part 2:", solve(input, 3))
}
